### DISCUSSION TITLE
What about using LSP4IJ?

### DISCUSSION THREAD

**Discussion by angelozerr** [snyk/snyk-intellij-plugin#563] [Ideas]
I noticed that you have implemented your own lsp support, but I would like to notify you that it exists now a free LSP support for IntelliJ https://github.com/redhat-developer/lsp4ij

 * You can see the current LSP support at https://github.com/redhat-developer/lsp4ij/blob/main/docs%2FLSPSupport.md

 * LSP4IJ provides an LSP console to show LSP traces and see the state of language server https://github.com/redhat-developer/lsp4ij/blob/main/docs%2FUserGuide.md

 * If you want to play quickly with LSP4IJ without developping an IJ plugin to evaluate it, you can do that with https://github.com/redhat-developer/lsp4ij/blob/main/docs/UserDefinedLanguageServer.md

 * If you want to integrate your ls in an IJ plugin please read https://github.com/redhat-developer/lsp4ij/blob/main/docs%2FDeveloperGuide.md

If you need any features, please create an issue at https://github.com/redhat-developer/lsp4ij/issues

You can read an article about LSP4IJ at https://idetools.dev/blog/lsp4ij-announcement/

Hope you could be interested with LSP4IJ and don't hesitate to answer me if you have any questions.

Thanks!

**Comment by bastiandoetsch** [comment]
Good job with that and thanks for pointing out that you're open for business :). When we were starting, it was not that far along and still part of the Quarkus code-base, but now, we'll maybe do a spike to check it out. It's loosely based on LSP4e, right?

**Reply by angelozerr** [reply]
>  It's loosely based on LSP4e, right?

The project has been created by copying / pasting code from LSP4E and adapt it for IJ, but today we don't copy/paste code from LSP4E.

**Reply by angelozerr** [reply]
@bastiandoetsch have you tried LSP4IJ? Since my last comment, LSP4IJ has been improved a lot and we have more and more adopters which uses LSP4IJ https://github.com/redhat-developer/lsp4ij/tree/main?tab=readme-ov-file#who-is-using-lsp4ij

**Comment by angelozerr** [comment]
@bastiandoetsch have you tried LSP4IJ?

**Reply by bastiandoetsch** [reply]
Hey @angelozerr , we haven't. So far, we didn't have the need to update our existent LSP4j code. As a plugin that already has the "glue" custom implemented, what would be the advantages you see right now? Personally, I doubt the effort has a tangible RoI right now. In the future that may change with updates and newer functionality that we haven't covered yet, though.

**Reply by angelozerr** [reply]
@bastiandoetsch if you are confident with your LSP support and you don't need to implement new LSP features, I understand.
